rest_framework_simplejwt/serializers.py

```diff
@@ -105,27 +106,51 @@ class TokenRefreshSerializer(serializers.Serializer):
     token_class = RefreshToken
 
     default_error_messages = {
-        "no_active_account": _("No active account found for the given token.")
+        "no_active_account": _("No active account found for the given token."),
+        "password_changed": _("The user's password has been changed."),
     }
 
     def validate(self, attrs: dict[str, Any]) -> dict[str, str]:
         refresh = self.token_class(attrs["refresh"])
 
         user_id = refresh.payload.get(api_settings.USER_ID_CLAIM, None)
         if user_id:
-            user = (
-                get_user_model()
-                .objects.filter(**{api_settings.USER_ID_FIELD: user_id})
-                .first()
-            )
-            if not user or not api_settings.USER_AUTHENTICATION_RULE(user):
+            try:
+                user = get_user_model().objects.get(
+                    **{api_settings.USER_ID_FIELD: user_id}
+                )
+            except get_user_model().DoesNotExist:
+                # This handles the case where the user has been deleted.
                 raise AuthenticationFailed(
-                    self.error_messages["no_active_account"],
-                    "no_active_account",
+                    self.error_messages["no_active_account"], "no_active_account"
                 )
 
-        data = {"access": str(refresh.access_token)}
+            if not api_settings.USER_AUTHENTICATION_RULE(user):
+                raise AuthenticationFailed(
+                    self.error_messages["no_active_account"], "no_active_account"
+                )
```

Comment:
rebase properly. I think your changes are the try/except; please ensure it uses the code from #861